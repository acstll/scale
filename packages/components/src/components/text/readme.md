# t-text



<!-- Auto Generated Below -->


## Properties

| Property      | Attribute      | Description          | Type     | Default |
| ------------- | -------------- | -------------------- | -------- | ------- |
| `customClass` | `custom-class` | (optional) Tag class | `string` | `''`    |
| `theme`       | `theme`        | (optional) Tag theme | `string` | `''`    |


----------------------------------------------

*Built with [StencilJS](https://stenciljs.com/)*
