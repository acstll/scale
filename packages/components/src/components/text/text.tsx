import { Component, h, Prop } from '@stencil/core';
import { CssClassMap } from '../../utils/utils';
import classNames from 'classnames';

@Component({
  tag: 't-text',
  styleUrl: 'text.css',
  shadow: true,
})
export class Text {
  /** (optional) Tag class */
  @Prop() public customClass?: string = '';
  /** (optional) Tag theme */
  @Prop() public theme?: string = '';

  public render() {
    return (
      <div class={this.getCssClassMap()}>
        <slot />
      </div>
    );
  }

  private getCssClassMap(): CssClassMap {
    return classNames(
      'text',
      this.customClass && this.customClass,
      this.theme && `text--theme-${this.theme}`
    );
  }
}
