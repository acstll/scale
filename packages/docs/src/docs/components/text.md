---
title: Text
description: Text element
url: /docs/text
contributors:
  - eeegor
---

# Text

Text component

selector: `t-text`

## Default

```javascript
<t-text>Hello</t-text>
```

<div class="demo-container">
  <t-text>Hello</t-text>
</div>
