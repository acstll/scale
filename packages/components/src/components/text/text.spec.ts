import { newSpecPage } from '@stencil/core/testing';
import { Text } from './text';

describe('Text', () => {
  let element;
  beforeEach(async () => {
    element = new Text();
  });

  it('should match snapshot', async () => {
    const page = await newSpecPage({
      components: [Text],
      html: `<t-text>Label</t-text>`,
    });
    expect(page.root).toMatchSnapshot();
  });

  it('should handle a custom css class', () => {
    element.customClass = 'custom';
    expect(element.getCssClassMap()).toContain('custom');
  });

  it('should handle theme css class', () => {
    element.theme = 'default';
    expect(element.getCssClassMap()).toContain('text--theme-default');
  });

  it('should have a default css class', () => {
    expect(element.getCssClassMap()).toContain('text');
  });
});
