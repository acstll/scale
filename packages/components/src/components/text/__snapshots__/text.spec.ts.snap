// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match snapshot 1`] = `
<t-text>
  <mock:shadow-root>
    <div class="text">
      <slot></slot>
    </div>
  </mock:shadow-root>
  Label
</t-text>
`;
